echopop/tests/test_kriging_biomass_apportionment.py

```diff
@@ -0,0 +1,878 @@
+        }
+    )
+
+    ### Mock data for `weight_strata`
+    test_weight_strata = pd.DataFrame(
+        {
+            'stratum_num': [ 0 , 1 ] ,
+            'weight_stratum_all': [ 50.0 , 100.0 ] ,
+        }
+    )
+
+    ### Evaluate for later comparison 
+    eval_aged_unaged_weight_proportions = compute_aged_unaged_proportions( test_specimen_data ,
+                                                                           test_weight_strata )
+    
+    ###--------------------------------
+    ### Expected outcomes
+    ###--------------------------------
+    ### `eval_aged_sex_proportions`
+    # ---- Expected dimensions
+    expected_dimensions = tuple( [ 2 , 7 ] )
+    # ---- Expected dataframe output
+    expected_output = pd.DataFrame( {
+        'stratum_num': np.array( [ 0.0 , 1.0 ] ).astype( np.int64 ) ,
+        'weight': [ 20.0 , 20.0 ] ,
+        'weight_stratum_all': [ 50.0 , 100.0 ] ,
+        'proportion_aged_weight_all': [ 0.40 , 0.20 ] ,
+        'proportion_aged_weight_adult': [ 0.30 , 0.15 ] ,
+        'proportion_unaged_weight_all': [ 0.60 , 0.80 ] ,
+        'proportion_unaged_weight_adult': [ 0.70 , 0.85 ]
+    } )
+
+    #----------------------------------
+    ### Run tests: `compute_index_aged_weight_proportions`
+    #----------------------------------
+    ### `eval_aged_sex_proportions`
+    # ---- Shape
+    assert eval_aged_unaged_weight_proportions.shape == expected_dimensions
+    # ---- Datatypes
+    assert np.all( eval_aged_unaged_weight_proportions.dtypes == expected_output.dtypes )
+    # ---- Dataframe equality
+    assert np.allclose( eval_aged_unaged_weight_proportions , expected_output )
+
+def test_compute_aged_sex_proportions( ):
+
+    ### Mock data for `proportions_weight_length_age_sex`
+    test_proportions_weight_length_age_sex = pd.DataFrame( {
+        'stratum_num': np.repeat( [ 0 , 1 ] , 8 ).astype( np.int64 ) ,
+        'species_id': np.repeat( 19350 , 16 ).astype( np.int64 ) ,
+        'sex': np.tile( [ 'female' , 'female' , 'female' , 'female' , 
+                        'male' , 'male' , 'male' , 'male'] , 2 ).astype( object ).flatten( ) ,
+        'length_bin': pd.cut( np.tile( [ 12 , 12 , 19 , 19 ] , 4 ) , 
+                            np.linspace( 9 , 21 , 3 ) ) ,
+        'age_bin': pd.cut( np.tile( [ 1 , 2 ] , 8 ) , 
+                        np.array( [ 0.5 , 1.5 , 2.5 ] ) ) ,
+        'weight_all': [ 3.0 , 0.0 , 0.0 , 7.0 , 2.0 , 0.0 , 0.0 , 8.0 ,
+                        4.0 , 0.0 , 0.0 , 6.0 , 1.0 , 0.0 , 0.0 , 9.0 ] ,
+        'weight_adult': [ 0.0 , 0.0 , 0.0 , 7.0 , 0.0 , 0.0 , 0.0 , 8.0 ,
+                        0.0 , 0.0 , 0.0 , 6.0 , 0.0 , 0.0 , 0.0 , 9.0 ] ,
+        'total_weight_sex_all': np.repeat( 10 , 16 ).astype( np.float64 ).flatten( ) ,
+        'total_weight_sex_adult': [ 7.0 , 7.0 , 7.0 , 7.0 , 8.0 , 8.0 , 8.0 , 8.0 ,
+                                    6.0 , 6.0 , 6.0 , 6.0 , 9.0 , 9.0 , 9.0 , 9.0 ] ,
```

Comment:
I am confused about what these quantities are -- maybe it's time to put together a table for variation/column names in different part of the data structure? 
I think I am confused because `weight_adult` is 0 for the second row, but `total_weight_sex_adult` has a non-zero value. So I felt this goes back to I do not know what these quantities are.